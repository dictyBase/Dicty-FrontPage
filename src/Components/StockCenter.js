// @flow
import React from 'react';
import FontAwsome from 'react-fontawesome';

import {
  StockTitle,
  StockHeader,
  StockSubHeader,
  ListBox,
  StockBox,
  AnnotationListItems,
  StockContainer,
  MoreLink,
  NewsStockTitle
} from './styles';

const StockCenter = props => {
  const plasmidlist = props.stockcenter.plasmids.map((plasmid, index) =>
    <AnnotationListItems key={index}>
      {plasmid}
    </AnnotationListItems>
  );

  const strainlist = props.stockcenter.strains.map((strain, index) =>
    <AnnotationListItems key={index}>
      {strain}
    </AnnotationListItems>
  );

  return (
    <StockContainer>
      <StockHeader>
        <FontAwsome name="shopping-cart fa-md" />
        <NewsStockTitle>DICTY STOCK CENTER</NewsStockTitle>
      </StockHeader>
      <StockSubHeader>New items</StockSubHeader>
      <StockBox>
        <StockTitle>PLASMIDS</StockTitle>
        <ListBox margintop="5px" padbottom="0px">
          {plasmidlist}
        </ListBox>
        <MoreLink padbottom="0px">
          <FontAwsome name="plus fa-xs" />
        </MoreLink>
      </StockBox>
      <StockBox color="white" background={'#0489B1'}>
        <StockTitle>STRAINS</StockTitle>
        <ListBox margintop="5px" padbottom="0px">
          {strainlist}
        </ListBox>
        <MoreLink padbottom="0px">
          <FontAwsome name="plus fa-xs" />
        </MoreLink>
      </StockBox>
    </StockContainer>
  );
};

export default StockCenter;
