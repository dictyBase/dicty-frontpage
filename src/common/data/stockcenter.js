export default {
  plasmids: ["pTgrR1/LacZ", "pBeiB/YFP", "pSigK", "pExpl7/lacZ"],
  strains: ["Mad52", "grlD-/[act15]:grlD", "grlD-", "ase1A-"],
}
