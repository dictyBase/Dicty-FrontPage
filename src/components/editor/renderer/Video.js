import React, { PureComponent } from "react"

export default class Video extends PureComponent {
  onClick = e => {
    e.stopPropagation()
  }

  render() {
    const { node, isSelected, editor, attributes } = this.props
    const src = node.data.get("src")
    const url = node.data.get("url")

    const { readOnly } = editor.props

    const wrapperStyle = {
      position: "relative",
      paddingBottom: "50.66%",
      paddingTop: "25px",
      height: "0",
      outline: isSelected ? `2px solid #0017ff` : "none",
    }

    const maskStyle = {
      display: isSelected ? "none" : "block",
      position: "absolute",
      top: "0px",
      left: "0px",
      right: "0px",
      bottom: "0px",
      height: "100%",
      cursor: "cell",
      zIndex: 1,
    }

    const iframeStyle = {
      position: "absolute",
      top: "0px",
      left: "0px",
      width: "100%",
      height: "100%",
    }

    return (
      <div>
        <div {...attributes} style={wrapperStyle}>
          <div style={readOnly ? {} : maskStyle} />
          <iframe
            title="video-embed"
            id="ytplayer"
            type="text/html"
            width="580"
            height="390"
            src={src}
            frameBorder="0"
            style={iframeStyle}
          />
        </div>
      </div>
    )
  }
}
