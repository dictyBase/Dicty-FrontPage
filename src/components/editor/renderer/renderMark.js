// @flow
import React, { type Node } from "react"
import { type Mark } from "slate"
import MARKS from "components/editor/constants/marks"

type Props = {
  children: Node,
  mark: Mark,
}

const renderMark = (props: Props) => {
  const { children, mark } = props
  switch (mark.type) {
    case MARKS.BOLD:
      return <strong>{children}</strong>

    case MARKS.ITALIC:
      return <em>{children}</em>

    case MARKS.UNDERLINE:
      return <u>{children}</u>

    case MARKS.STRIKETHROUGH:
      return <del>{children}</del>

    default:
      return null
  }
}

export default renderMark
