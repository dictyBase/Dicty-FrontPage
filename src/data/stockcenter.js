export default {
  plasmids: ["DBS0304455", "DSB0350408", "DBS0350326"],
  strains: ["DBS0304455", "DSB0350408", "DBS0350326", "DBS0350435"],
}
