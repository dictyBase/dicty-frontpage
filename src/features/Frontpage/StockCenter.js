import React from "react"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"

const styles = (theme) => ({
  listItem: {
    listStyle: "none",
    fontSize: "12px",
  },
  listBox: {
    padding: "0px 25px 10px 25px",
    fontSize: "12px",
    marginBottom: "5px",
    paddingBottom: "0px",
    marginTop: "5px",
    "@media (max-width: 992px) and (min-width: 767px)": {
      fontSize: "10px",
    },
    "@media (max-width: 768px)": {
      fontSize: "16px",
    },
  },
  plusSign: {
    color: "#0b3861",
    fontSize: "11px",
    fontStyle: "italic",
    fontWeight: "normal",
    textAlign: "center",
    paddingBottom: "0px",

    "@media (min-width: 1400px)": {
      paddingTop: "30px",
      fontSize: "12px",
    },
  },
  title: {
    paddingTop: "5px",
    paddingLeft: "5px",
  },
  subheader: {
    backgroundColor: "#effbfb",
    color: "black",
    fontSize: "12px",
    letterSpacing: "5px",
    padding: "1px 0px 1px 0px",

    "@media (max-width: 767px)": {
      fontSize: "12px",
      textAlign: "center",
    },
  },
  header: {
    backgroundColor: "#004080",
    color: "#efeffb",
    fontSize: "18px",
    textAlign: "center",
    padding: "10px 0px 10px 0px",
    border: "1px dotted #f5f6ce",
    borderTopLeftRadius: "4px",
    borderTopRightRadius: "4px",
    "@media (max-width: 767px)": {
      fontSize: "18px",
      textAlign: "center",
    },
  },
  container: {
    textAlign: "center",
    padding: "5px 5px 0 5px",
    marginBottom: "10px",
  },
  plasmidBox: {
    color: "#fff",
    backgroundColor: "#0073e6",
    textAlign: "center",
    paddingBottom: "5px",
    borderBottomLeftRadius: "4px",
    borderBottomRightRadius: "4px",
    "@media (max-width: 992px) and (min-width: 767px)": {
      fontSize: "10px",
    },
    "@media (max-width: 767px)": {
      fontSize: "12px",
    },
  },
  strainBox: {
    color: "#242124",
    backgroundColor: "#80c1ff",
    textAlign: "center",
    paddingBottom: "5px",
    borderBottomLeftRadius: "4px",
    borderBottomRightRadius: "4px",
    "@media (max-width: 992px) and (min-width: 767px)": {
      fontSize: "10px",
    },
    "@media (max-width: 767px)": {
      fontSize: "12px",
    },
  },
})

/** Widget that displays the most recent plasmids and strains in the Stock Center */

const StockCenter = (props) => {
  const { classes, stockcenter } = props

  const plasmidlist = stockcenter.plasmids.map((plasmid, index) => (
    <li className={classes.listItem} key={index}>
      {plasmid}
    </li>
  ))

  const strainlist = stockcenter.strains.map((strain, index) => (
    <li className={classes.listItem} key={index}>
      {strain}
    </li>
  ))

  return (
    <div className={classes.container}>
      <div className={classes.header}>
        <FontAwesomeIcon icon="shopping-cart" size="sm" />
        <span className={classes.title}> DICTY STOCK CENTER</span>
      </div>
      <div className={classes.subheader}>
        <strong>New items</strong>
      </div>
      <Grid container>
        <Grid item xs={6} className={classes.plasmidBox}>
          <div className={classes.title}>PLASMIDS</div>
          <ul className={classes.listBox}>{plasmidlist}</ul>
          <div className={classes.plusSign}>
            <FontAwesomeIcon icon="plus" />
          </div>
        </Grid>
        <Grid item xs={6} className={classes.strainBox}>
          <div className={classes.title}>STRAINS</div>
          <ul className={classes.listBox}>{strainlist}</ul>
          <div className={classes.plusSign}>
            <FontAwesomeIcon icon="plus" />
          </div>
        </Grid>
      </Grid>
    </div>
  )
}

export default withStyles(styles)(StockCenter)
