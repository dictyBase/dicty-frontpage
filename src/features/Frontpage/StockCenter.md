```js
import stockcenter from "data/stockcenter"

;<StockCenter stockcenter={stockcenter} />
```
