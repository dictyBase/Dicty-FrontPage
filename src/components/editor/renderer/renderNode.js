// @flow
import React, { type Node } from "react"
import { type Block } from "slate"
import styled from "styled-components"
import Video from "components/editor/renderer/Video"
import BLOCKS from "components/editor/constants/blocks"
import INLINES from "components/editor/constants/inlines"

type Props = {
  node: Block,
  children: Node,
  attributes: Object,
}

const Image = styled.img`
  display: block;
  max-width: 100%;
  max-height: 20em;
  box-shadow: ${props => (props.selected ? "0 0 0 2px blue;" : "none")};
`

const renderNode = (props: Props) => {
  const { attributes, children, node } = props
  switch (node.type) {
    case BLOCKS.UL_LIST:
      return <ul {...attributes}>{children}</ul>

    case BLOCKS.OL_LIST:
      return <ol {...attributes}>{children}</ol>

    case BLOCKS.LIST_ITEM:
      return <li {...attributes}>{children}</li>

    case BLOCKS.HEADING_1:
      return <h1 {...attributes}>{children}</h1>

    case BLOCKS.HEADING_2:
      return <h2 {...attributes}>{children}</h2>

    case BLOCKS.HEADING_3:
      return <h3 {...attributes}>{children}</h3>

    case BLOCKS.BLOCKQUOTE:
      return <blockquote {...attributes}>{children}</blockquote>

    case INLINES.LINK: {
      const { data } = node
      const href = data.get("href")
      return (
        <a {...attributes} href={href}>
          {children}
        </a>
      )
    }

    case BLOCKS.IMAGE: {
      const { data, isSelected } = node
      const src = data.get("src")
      const className = isSelected ? "selected" : "unselected"
      const style = { display: "inline-block" }
      return (
        <img
          src={src}
          className={className}
          {...attributes}
          style={style}
          alt=""
        />
      )
    }

    case BLOCKS.ALIGN_LEFT:
      return (
        <div style={{ textAlign: "left" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_CENTER:
      return (
        <div style={{ textAlign: "center" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_RIGHT:
      return (
        <div style={{ textAlign: "right" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.HR:
      return <hr {...attributes} />

    case BLOCKS.TABLE:
      return (
        <table {...attributes}>
          <tbody>{children}</tbody>
        </table>
      )

    case BLOCKS.TABLE_ROW:
      return <tr {...attributes}>{children}</tr>

    case BLOCKS.TABLE_CELL:
      return <td {...attributes}>{children}</td>

    case BLOCKS.VIDEO:
      return <Video {...props} />

    default:
      return null
  }
}

export default renderNode
