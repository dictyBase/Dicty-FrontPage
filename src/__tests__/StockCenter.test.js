import React from "react"
import StockCenter from "../Components/frontpage/StockCenter"
import renderer from "react-test-renderer"
import stockcenter from "../././data/stockcenter"

test("StockCenter snapshot test", () => {
  const component = renderer.create(<StockCenter stockcenter={stockcenter} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
