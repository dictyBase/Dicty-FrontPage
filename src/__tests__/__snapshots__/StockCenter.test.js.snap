// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockCenter snapshot test 1`] = `
<div
  className="sc-bxivhb dKDGug"
>
  <div
    className="sc-EHOje glvWVV"
  >
    <span
      aria-hidden={true}
      className="fa fa-shopping-cart fa-md"
    />
    <span
      className="sc-jKJlTe ddVBtl"
    >
      DICTY STOCK CENTER
    </span>
  </div>
  <div
    className="sc-bZQynM leuWTL"
  >
    New items
  </div>
  <div
    className="sc-cSHVUG kmYtrK"
  >
    <div
      className="sc-gzVnrw lfEvdn"
    >
      PLASMIDS
    </div>
    <ul
      className="sc-jzJRlG bSWWzu"
    >
      <li
        className="sc-iwsKbI dukjNo"
      >
        DBS0304455
      </li>
      <li
        className="sc-iwsKbI dukjNo"
      >
        DSB0350408
      </li>
      <li
        className="sc-iwsKbI dukjNo"
      >
        DBS0350326
      </li>
    </ul>
    <div
      className="sc-kgoBCf fagbkO"
    >
      <span
        aria-hidden={true}
        className="fa fa-plus fa-xs"
      />
    </div>
  </div>
  <div
    className="sc-cSHVUG WqFJo"
    color="white"
  >
    <div
      className="sc-gzVnrw lfEvdn"
    >
      STRAINS
    </div>
    <ul
      className="sc-jzJRlG bSWWzu"
    >
      <li
        className="sc-iwsKbI dukjNo"
      >
        DBS0304455
      </li>
      <li
        className="sc-iwsKbI dukjNo"
      >
        DSB0350408
      </li>
      <li
        className="sc-iwsKbI dukjNo"
      >
        DBS0350326
      </li>
      <li
        className="sc-iwsKbI dukjNo"
      >
        DBS0350435
      </li>
    </ul>
    <div
      className="sc-kgoBCf fagbkO"
    >
      <span
        aria-hidden={true}
        className="fa fa-plus fa-xs"
      />
    </div>
  </div>
</div>
`;
